bookkeeper-server/src/test/java/org/apache/bookkeeper/bookie/CookieIndexDirTest.java

```diff
@@ -0,0 +1,973 @@
+        String journalDir = newDirectory();
+        ServerConfiguration conf = TestBKConfiguration.newServerConfiguration();
+        conf.setJournalDirName(journalDir)
+                .setLedgerDirNames(ledgerDirs)
+                .setIndexDirName(indexDirs)
+                .setBookiePort(bookiePort)
+                .setMetadataServiceUri(zkUtil.getMetadataServiceUri());
+
+        validateConfig(conf);
+
+        File cookieFile =
+                new File(BookieImpl.getCurrentDirectory(new File(indexDirs[0])), BookKeeperConstants.VERSION_FILENAME);
+        assertTrue(cookieFile.delete());
+        try {
+            validateConfig(conf);
+            fail("Shouldn't have been able to start");
+        } catch (InvalidCookieException ice) {
+            // correct behaviour
+        }
+    }
+
+    /**
+     * Test that if a ledger directory is added to a
+     * preexisting bookie, the bookie will fail
+     * to start.
+     */
+    @Test
+    public void testLedgerDirectoryAdded() throws Exception {
+        String ledgerDir0 = newDirectory();
+        String indexDir0 = newDirectory();
+        String journalDir = newDirectory();
+        ServerConfiguration conf = TestBKConfiguration.newServerConfiguration();
+        conf.setJournalDirName(journalDir)
+            .setLedgerDirNames(new String[] { ledgerDir0 })
+            .setIndexDirName(new String[] { indexDir0 })
+            .setBookiePort(bookiePort)
+            .setMetadataServiceUri(zkUtil.getMetadataServiceUri());
+
+        validateConfig(conf);
+
+        conf.setLedgerDirNames(new String[] { ledgerDir0, newDirectory() });
+        try {
+            validateConfig(conf);
+            fail("Shouldn't have been able to start");
+        } catch (InvalidCookieException ice) {
+            // correct behaviour
+        }
+
+        conf.setLedgerDirNames(new String[] { ledgerDir0 });
+        validateConfig(conf);
+    }
+
+    /**
+     * Test that if a index directory is added to a
+     * preexisting bookie, the bookie will fail
+     * to start.
+     */
+    @Test
+    public void testIndexDirectoryAdded() throws Exception {
+        String ledgerDir0 = newDirectory();
+        String indexDir0 = newDirectory();
+        String journalDir = newDirectory();
+        ServerConfiguration conf = TestBKConfiguration.newServerConfiguration();
+        conf.setJournalDirName(journalDir)
+                .setLedgerDirNames(new String[] { ledgerDir0 })
+                .setIndexDirName(new String[] { indexDir0 })
+                .setBookiePort(bookiePort)
+                .setMetadataServiceUri(zkUtil.getMetadataServiceUri());
+
+        validateConfig(conf);
+
+        conf.setIndexDirName(new String[] { indexDir0, newDirectory() });
+        try {
+            validateConfig(conf);
+            fail("Shouldn't have been able to start");
+        } catch (InvalidCookieException ice) {
+            // correct behaviour
```

Comment:
is there a way to add a better assertion here (and in other similar points) ? it may have failed for other reasons

Reply:
> is there a way to add a better assertion here (and in other similar points) ? it may have failed for other reasons

here is expected to throw InvalidCookieException, if not, fail this testcase.
so InvalidCookieException is enough, I think @eolivelli